webhook-service/lib/ip_resolver.go

```diff
@@ -0,0 +1,42 @@
+package lib
+
+import (
+	"net"
+	"net/url"
+
+	logger "github.com/sirupsen/logrus"
+)
+
+type IPResolver interface {
+	Resolve(curlURL string) []string
+}
+
+type LookupFunc func(host string) ([]net.IP, error)
+
+type IpResolver struct {
+	LookupIP LookupFunc
+}
+
+func NewIPResolver(lookUpIPFunc ...LookupFunc) IPResolver {
+	return IpResolver{
+		LookupIP: net.LookupIP,
+	}
+}
+
+func (i IpResolver) Resolve(curlURL string) []string {
```

Comment:
```suggestion
func (i IpResolver) Resolve(url string) []string {
```

Reply:
since the url is given as a raw string, you should also test correct behavior of the method when feeding it with an invalid url....